app/src/main/java/com/locoquest/app/MainActivity.kt

```diff
@@ -198,15 +201,56 @@ class MainActivity : AppCompatActivity(), OnMapReadyCallback {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-    // Find the FragmentContainerView that contains the map
+        // Initialize Google Map
+        initGoogleMap()
+
+        // Initialize signInButton
+        signInButton = findViewById(R.id.google_sign_in_button)
+
+        // Firebase Sign-in
+        oneTapClient = Identity.getSignInClient(this)
+        signUpRequest = BeginSignInRequest.builder()
+            .setGoogleIdTokenRequestOptions(
+                BeginSignInRequest.GoogleIdTokenRequestOptions.builder()
+                    .setSupported(true)
+                    // Your server's client ID, not your Android client ID.
+                    .setServerClientId(getString(R.string.web_client_id))
+                    // Show all accounts on the device.
+                    .setFilterByAuthorizedAccounts(false)
+                    .build())
+            .build()
+
+        // Set the OnClickListener for signInButton
+        signInButton.setOnClickListener {
+            oneTapClient.beginSignIn(signUpRequest)
+                .addOnSuccessListener(this) { result ->
+                    try {
+                        startIntentSenderForResult(
+                            result.pendingIntent.intentSender, REQ_ONE_TAP,
+                            null, 0, 0, 0
+                        )
+                    } catch (e: IntentSender.SendIntentException) {
+                        Log.e(TAG, "Couldn't start One Tap UI: ${e.localizedMessage}")
+                    }
+                }
+                .addOnFailureListener(this) { e ->
+                    // No Google Accounts found. Just continue presenting the signed-out UI.
+                    Log.d(TAG, e.localizedMessage)
+                }
+        }
+    }
+
+
+    private fun initGoogleMap() {
```

Comment:
Moved out the initializing map from the OnCreate method to reduce clutter

https://kotlinlang.org/docs/functions.html